src/main/host/descriptor/mod.rs

```diff
@@ -32,6 +32,17 @@ bitflags::bitflags! {
     }
 }
 
+impl FileStatus {
+    pub fn as_o_flags(&self) -> libc::c_int {
```

Comment:
Maybe worth using `nix::fcntl::OFlag` instead of bare ints, throughout? Or too much conversion boilerplate when we do need to get to a bare int?

Reply:
Thanks, I didn't know about that. I added it in and it seems like an improvement.